File: backend/app/models/quote_item.py
from decimal import Decimal
from typing import TYPE_CHECKING

from sqlalchemy import ForeignKey, Numeric
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base_model import BaseModel

if TYPE_CHECKING:
    from app.models.quote import Quote
    from app.models.product import Product


class QuoteItem(BaseModel):
    __tablename__ = "quote_items"

    quote_id: Mapped[int] = mapped_column(ForeignKey("quotes.id"))

    product_id: Mapped[int] = mapped_column(ForeignKey("products.id"))

    quantity: Mapped[int] = mapped_column(nullable=False)

    unit_price: Mapped[Decimal] = mapped_column(
        Numeric(10, 2),
        nullable=False,
    )

    discount: Mapped[Decimal] = mapped_column(
        Numeric(10, 2),
        default=Decimal("0.00"),
        nullable=False,
    )

    quote: Mapped["Quote"] = relationship(back_populates="items")
    product: Mapped["Product"] = relationship()

File: frontend/src/app/features/quotes/models/quote-item.ts
import { ProductSummary } from '@features/products/models/product-summary';
import { QuoteItemTotals } from './quote-item-totals';

export interface QuoteItem {
  id: number;
  quoteId: number;
  productId: number;

  product: ProductSummary;

  quantity: number;
  unitPrice: number;
  discount: number;

  totals: QuoteItemTotals;

  createdAt: string;
  updatedAt: string;
}
